### PR TITLE
OCT-688 - Update aws-sdk to v3

### PR DESCRIPTION
**Pull request by Harry-Jisc**
The purpose of this PR was to upgrade from aws-sdk v2 to v3.

---

### Acceptance Criteria:

As Per [OCT-688](https://jiscdev.atlassian.net/browse/OCT-688?atlOrigin=eyJpIjoiZGRhM2U1MzYwNmZiNGRhOGFkNTAxMDY5NmNiZjZkZTEiLCJwIjoiaiJ9)

---

### Checklist:

- [x] Local manual testing conducted
- [x] Automated tests conducted
---
### Screenshots:
Screenshot 2023-06-28 at 15 25 08

[OCT-688]: https://jiscdev.atlassian.net/browse/OCT-688?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: api/package.json [hunk 1]
```diff
@@ -66,6 +63,9 @@
         "supertest": "^6.2.3"
     },
     "devDependencies": {
+        "@aws-sdk/client-s3": "^3.360.0",
```
Comment (florin-holhos) [inline api/package.json:66]: image

all these packages must be excluded from the bundle with 'forceExclude' in serverless.yml
You can take quick a look on my PR to see an example how i excluded puppeteer https://github.com/JiscSD/octopus/pull/430/files#:~:text=includeModules%3A-,forceExclude,-%3A%20puppeteer

File: api/src/components/publication/controller.ts [hunk 2]
```diff
@@ -378,6 +377,8 @@ export const getPDF = async (
     const redirectToPreview = event.queryStringParameters?.redirectToPreview === 'true';
     const publicationId = event.pathParameters.id;
     const publication = await publicationService.get(publicationId);
+    const s3Endpoint =
+        process.env.STAGE === 'local' ? process.env.LOCALSTACK_SERVER : 'https://s3.eu-west-1.amazonaws.com';
```
Comment (florin-holhos) [inline api/src/components/publication/controller.ts:381]: This is not needed. You can import s3 from /lib/s3.ts and then just use {s3.config.endpoint}

File: api/src/components/publication/controller.ts [hunk 3]
```diff
@@ -396,7 +397,7 @@ export const getPDF = async (
     if (!generateNewPDF) {
         // check if there's a generated PDF for this publication
         try {
-            const currentPdfUrl = `${s3.endpoint.href}science-octopus-publishing-pdfs-${process.env.STAGE}/${publicationId}.pdf`;
+            const currentPdfUrl = `${s3Endpoint}/science-octopus-publishing-pdfs-${process.env.STAGE}/${publicationId}.pdf`;
```
Comment (florin-holhos) [inline api/src/components/publication/controller.ts:400]: same as above, the endpoint needs to come from the s3 config

File: api/src/components/publication/service.ts [hunk 4]
```diff
@@ -742,6 +744,8 @@ export const getLinksForPublication = async (id: string) => {
 export const generatePDF = async (publication: I.Publication & I.PublicationWithMetadata): Promise<string | null> => {
     const references = await referenceService.getAllByPublication(publication.id);
     const htmlTemplate = Helpers.createPublicationHTMLTemplate(publication, references);
+    const s3Endpoint =
```
Comment (florin-holhos) [inline api/src/components/publication/service.ts:747]: same as above, the endpoint needs to come from s3 config

**Review by finlay-jisc — APPROVED** [review]

**Pull request merged**